server/snapshot/snapshot.go

```diff
@@ -0,0 +1,161 @@
+package snapshot
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/Sirupsen/logrus"
+
+	"github.com/docker/notary/server/storage"
+	"github.com/docker/notary/tuf/data"
+	"github.com/docker/notary/tuf/signed"
+)
+
+// GetOrCreateSnapshotKey either creates a new snapshot key, or returns
+// the existing one. Only the PublicKey is returned. The private part
+// is held by the CryptoService.
+func GetOrCreateSnapshotKey(gun string, store storage.KeyStore, crypto signed.CryptoService, createAlgorithm string) (data.PublicKey, error) {
```

Comment:
I'm not sure where it'd go, but could this be unified with `GetOrCreateTimestampKey`?  The code seems the same except for the role.